README.md

```diff
@@ -508,6 +509,46 @@ testcases:
 
 ```
 
+## Iterating over data
+
+It is possible to iterate over data using `range` attribute.
+
+The following data types are supported, each exposing contexted variables `.index`, `.key` and `.value`:
+
+- An array where each value will be iterated over (`[]interface{}`)
+  - `.index`/`.key`: current iteration index
+  - `.value`: current iteration item value
+- A map where each key will be iterated over (`map[string]interface{}`)
+  - `.index`: current iteration index
+  - `.key`: current iteration item key
+  - `.value`: current iteration item value
+- An integer to perform target step `n` times (`int`)
+  - `.index`/`.key`/`.value`: current iteration index
+- A templated string which results in one of the above typing (`string`)
+  - It can be either inherited from vars file, or interpolated from a previous step result
+
+Within a range context, `.range` contexted variable will be set to `true` (while it'll be `false` in regular contexts)
+
+For instance, the following example will iterate over an array of two items containing maps:
+```yaml
+- name: range with harcoded array
+  steps:
+  - type: exec
+    range:
+      - actual: hello
+        expected: hello
+        index: 0
```

Comment:
avoid this field in the map example, it can be confusing to have `value.index` and `.index`

Reply:
it's mostly for testing that `.index` is effectively templated and set to correct value, but from an user pov it makes no sense
maybe just renaming `index` to `testIndex` would be better ?

Reply:
it's fine to keep it in our tests files. but avoid it in the README.md

Reply:
ok I removed it from `README.md`